src/bin/tikv-server.rs

```diff
@@ -181,24 +217,66 @@ fn main() {
         print_usage(&program, opts);
         return;
     }
-    initial_log(&matches);
 
-    let addr = matches.opt_str("A").unwrap_or_else(|| DEFAULT_LISTENING_ADDR.to_owned());
+    let config = match matches.opt_str("C") {
+        Some(path) => {
+            let mut config_file = fs::File::open(&path).expect("config open filed");
+            let mut s = String::new();
+            config_file.read_to_string(&mut s).expect("config read filed");
+            toml::Value::Table(toml::Parser::new(&s).parse().expect("malformed config file"))
+        }
+        // Empty value, lookup() always return `None`.
+        None => toml::Value::Integer(0),
+    };
+
+    initial_log(&matches, &config);
+
+    let addr = matches.opt_str("A").unwrap_or_else(|| {
```

Comment:
I had tried, but found it is difficult to implement.

1. Different values have different types,  `toml::Value::{String, Integer, Bool ...}`, which lead to `as_str(), as_integer(), as_bool() ...`

2. Different values have different behaves, eg: get `addr` from: command line > config file > default; get `pd` from: command line > config file > **panic!()** (no default value for pd)